src/util/FormUtils.js

```diff
@@ -478,4 +478,100 @@ export default class FormUtils {
       recursiveTraverse(q);
     });
   }
+
+  /**
+   * Performs a dfs on a set of questions and executes callback functions
+   * when entering and leaving each question.
+   * @param {Array<Object>} questions - The array of questions to traverse.
+   * @param {function} onEnterQuestion - The callback function to execute when entering a question.
+   * @param {function} onLeaveQuestion - The callback function to execute when leaving a question.
+   */
+  static dfsOnEnterAndLeaveQuestion(
```

Comment:
I suggest not having in the name "onEnterAndLeave" as it is clear from the parameters and DFS algorithmitself.

IMO, DFS algorithm itself clearly communicates entering and leaving the node ... e.g. by using stack (LIFO) where it holds nodes. It is strange to me to say it explicitly in the method name in the same matter as saying `fetchData()` ->  `fetchToReturnData()` or `deleteQuestion()` --> `deleteToNotHaveQuestion()`.

Before, the algorithm used onEnter question, which is IHO typically expected, but should be said explicitly, now, we just extended the algorithm.